lib/success.js

```diff
@@ -250,3 +243,32 @@ export default async function success(pluginConfig, context, { Octokit }) {
     throw new AggregateError(errors);
   }
 }
+
+/**
+ * Builds GraphQL query for fetching associated PRs to a list of commit hash (sha)
+ * @param {Array<string>} shas
+ * @returns {string}
+ */
+export function buildAssociatedPRsQuery(shas) {
+  return `#graphql
+    query getAssociatedPRs($owner: String!, $repo: String!) {
+      repository(owner: $owner, name: $repo) {
+        ${shas
+          .map((sha) => {
+            return `commit${sha.slice(0, 6)}: object(oid: "${sha}") {
+            ...on Commit {
+              associatedPullRequests(first: 100) {
+                nodes {
+                  url
+                  number
+                  body
+                }
+              }
+            }
+          }`;
+          })
+          .join("")}
+      }
+    }
```

Comment:
We will need to address the possibility that there might be more than 100 commits. If that occurs, we need to send multiple requests as only 100 nodes can be requested with a single GraphQL request

Reply:
was this addressed in this PR? im working with a repo that has more than 100 commits and am experiencing this secondary rate error on v24.0.0

Reply:
Hey @brandon-kyle-bailey

The Edge Case isn't addressed in this PR nor yet.

But, could you kindly open an issue with details of this error you're getting in regard this 100+ commit, thanks.

Reply:
unfortunately this is happening on an internal repo but ill do my best! @babblebey

Reply:
@babblebey im not able to reliable reproduce it at this point since re-running so will hold off on logging an issue. will revisit if it reappears!

Reply:
I'm currently working on a fix in regard the edge case here #892

Reply:
I'm currently working on a fix in regard the edge case here #892

Reply:
just capturing the relation to https://github.com/semantic-release/github/issues/867#issuecomment-2265881923 as well